packages/CodingStandard/src/Fixer/Property/ArrayPropertyDefaultValueFixer.php

```diff
@@ -70,8 +70,13 @@ private function isArrayPropertyDocComment(Token $token): bool
         return true;
     }
 
-    private function addDefaultValueForArrayProperty(Token $semicolonToken): void
+    private function addDefaultValueForArrayProperty(Tokens $tokens,int $semicolonPosition): void
     {
-        $semicolonToken->setContent(' = [];');
+        $tokens->insertAt($semicolonPosition, new Token(']'));
+        $tokens->insertAt($semicolonPosition, new Token('['));
+        $tokens->insertAt($semicolonPosition, new Token(' '));
```

Comment:
I know I enter this one a bit late, but I would like to point out one of the "difficulties" we face with tokens. I get it would be nice to override some tokens with new code in string form.
In your example you mention something like `[ ]`, lets say I would like to replace a token with some like `[$a]`.
Within PHP this can mean a lot of things, like;
`$a[$a];` -> array index
`[$a] = b();` -> array destruction on PHP 7.1 (CT tokens as PHP does not tokenize different)
`b = [$a];` -> new array with element one element; `$a`

In order to figure out how this code-string inserted should be tokenized to new tokens the context is needed or a lot back tracking has to be done. Therefor we did the trade off for performance over complex, slow and hard to maintain code.
 
So yes, dealing with Tokens does require some background knowledge about how PHP does it and how we worked around some of it within the fixer. This might not be perfect or under documented, so feel free to contribute or talk to us, as documenting it all is just to much we can handle atm.